uplink/ecclient/client.go

```diff
@@ -213,42 +213,35 @@ func (ec *ecClient) putPiece(ctx, parent context.Context, limit *pb.AddressedOrd
 		)
 		return nil, err
 	}
-	defer func() {
```

Comment:
Just a note: the main reason it was using defer instead of outside handling is that it's easy to forget to add the necessary cancel or commit in the right place, which would leak things.

Reply:
OK, but at least for me, I had to spend quite a little time to at the end understand the result of this function with this implementation.

If you think, that's worth it, I can take a look and see if we could have a `defer` just for only that purpose.

Let me know if I do that otherwise, I merge it right away.

Reply:
Yeah, I don't think I have a better solution.

Reply:
@egonelbre can you check the changes done in the last commit and see if you'd like more in that way?

It's keeping the `defer` for `cancel` and `commit` but I believe with fewer lines and easy to understand than the original one.

If you anyway see that this commit doesn't provide anything, let me know and I will revert it and merge the PR.

Reply:
Ah, yeah, that feels nicer than either of the previous.